packages/chat/src/components/AgentMessage/index.tsx

```diff
@@ -0,0 +1,68 @@
+import '../../styles.css';
+
+import clsx from 'clsx';
+import Markdown from 'react-markdown';
+import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
+import remarkGfm from 'remark-gfm';
+
+import Icon from '../Icon';
+import { aiIconModifier, contentStyle, generatedChin, messageContainer } from './AgentMessage.css';
+import codeTheme from './code-theme';
+
+interface IAgentMessage {
+  children: React.ReactNode;
+  from: 'system' | 'user';
+  generated?: boolean;
+  generatedMessage?: string;
+}
+
+export const AgentMessage: React.FC<IAgentMessage> = ({ children, generated, generatedMessage }) => {
+  const content = children?.toString();
+
+  const isCodeResponse =
+    (content?.startsWith('```') && (content?.endsWith('```') || content.endsWith('```\n'))) ||
+    content?.startsWith('\n```');
```

Comment:
you can probably use a single regex for all of this, as it stands this is a little hard to read
something kind of like `/^\s*```.*```\s*$/`

Reply:
I think `.*` in regex will not include `\n` which might be a problem.

Reply:
good catch, I think it should work if you make it a multiline regex `/^\s*```.*```\s*$/m`

Reply:
oh yeah you need an additional flag for it to match terminators, so I guess something like
`/^\s*```.*```\s*$/ms`